### PR TITLE
grt: ignore MIMCAP layers

### PR DESCRIPTION
**Pull request by eder-matheus**
The check for MIMCAP layers in Tcl will be added later after we have the swig interface for the LEF58_TYPE.

### CODE REVIEW

File: src/grt/src/GlobalRouter.cpp
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:2264]: Really getRoutingLayerCount & findRoutingLayer should handle this in the db as I don't think anybody wants to see mimcap as a routing layer.  Please move this to the db or let me know if you need help.

File: src/grt/src/RoutingTracks.h
Comment (maliberty) [inline src/grt/src/RoutingTracks.h:94]: Are these still needed now that you are using opendb layers?

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/grt/src/GlobalRouter.cpp
Comment (eder-matheus) [reply src/grt/src/GlobalRouter.cpp:2264]: Ok, this will make grt life easier. Besides all my updates to ignore the MIMCAP layers, I still have possible flaws related to pin layers, if they're placed at higher layers (like layer index 14 or something).

File: src/grt/src/RoutingTracks.h
Comment (eder-matheus) [reply src/grt/src/RoutingTracks.h:94]: I use them in some places where the function expects bools as arguments. I will update the function parameters.

**Comment by eder-matheus** [comment]
@maliberty rcx segaults with this update for 12nm testcases. I think the routing layer indices in rcx_FuncRCmin.rules should be updated. Who should contact about this issue?

**Comment by maliberty** [comment]
@sewkim can you help with regenerating the gf12 rcx rules file or at least provide Eder with instructions?

**Comment by eder-matheus** [comment]
@maliberty secure-ci is passing, except for bp_single (which is failing in the master branch too). Do you think we can merge this PR, or should we wait for a fix in secure-ci?

**Comment by maliberty** [comment]
I think we can merge it as there is no functional change here related to gf12.

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
@ravi-varadarajan this should enable you to use higher layers on gf12 once merged.

**Pull request merged**
